src/Pools/Group.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Utopia\Pools;
+
+use Exception;
+
+class Group
+{
+    /**
+     * @var Pool[]
+     */
+    protected array $pools = [];
+    
+    /**
+     * @var Pool $pool
+     * @return self
+     */
+    public function add(Pool $pool): self
+    {
+        $this->pools[$pool->getName()] = $pool;
+        return $this;
+    }
+    
+    /**
+     * @var string $name
+     * @return ?Pool
+     */
+    public function get(string $name): ?Pool
+    {
+        if(!isset($this->pools[$name])) {
+            throw new Exception('Pool not found');
+        }
+
+        return $this->pools[$name];
```

Comment:
```suggestion
        return $this->pools[$name] ??  throw new Exception('Pool not found');
```

This also works if you want it more short 👍🏻

Reply:
I always have this inner debate whether the shorter syntax is simple enough. Anyway, accepted the change.